src/sbvr-api/control-flow.ts

```diff
@@ -0,0 +1,60 @@
+import * as _ from 'lodash'
+import * as Promise from 'bluebird'
+
+// TODO: We force the return type here to merge `Promise<U> | Promise<U[]>` into `Promise<U | U[]>,
+// which is equivalent for our uses but fixes issues where typing were being rejected
+export const liftP = <T, U>(fn: (v: T) => U | Promise<U>): (a: T | T[]) => Promise<U | U[]> => {
+	return (a: T | T[]) => {
+		if (_.isArray(a)) {
+			// This must not be a settle as if any operation fails in a cs
+			// we want to discard the whole
+			return Promise.mapSeries(a, fn)
+		} else {
+			return Promise.resolve(a).then(fn)
+		}
+	}
+}
+
+interface MappingFunction {
+	<T, U>(a: T[], fn: (v: T) => U | Promise<U>): Promise<Array<U | Error>>
+}
+
+// The settle- versions of
```

Comment:
This comment looks like it’s partial?

Reply:
Yeah, I just left it completely as-is from the coffeescript, @nazrhom what is the comment meant to be?

Reply:
I changed it to ``The settle version of `Promise.mapSeries` `` which seems sane